mjml/core/registry.py

```diff
@@ -35,3 +39,26 @@ def _components():
 def assign_components(target, source):
     for component in source:
         target[component.component_name] = component
+
+
+def register_component(Component: Type):
+    assign_components(components, [Component])
+
+
+def register_custom_component(
```

Comment:
I don't really like this function as-is because it tries to do too much at once.

Compatibility with mjml upstream is mostly important for the XML dialect but not so much for API code when you integrate the Python module (though similar mechanisms might help).

- There should be one function which deals with a single component and (if necessary) one to register multiple components.
- I don't think callers should be able to override the registration function.
- Is "convert module into list of submodules" really an important use case which makes the API so much nicer? Ideally I'd just start with a really basic API which can be used but without any bells and whistles. Once this is in use we can extend the API to make it more convenient to use.

Stylistic issue if we keep the function: function signature on a single line, `comp` -> `component`.

Or maybe we can completely drop `register_custom_component()` and rename `assign_components()` to `register_components()`. No catching of `AttributeError`, `isinstance` checking?

Reply:
I think converting the module into a list of submodules is pretty convenient. I'm definitely open to modifying this for a cleaner implementation, though, if we decide to keep it. Developers can define all of their custom components in one module and then simply reference that when calling `mjml_to_html`.

I considered going the config file approach (like MJML upstream), but I thought this seemed simpler. That approach does have the benefit of being able to define custom components only once, without needing to specify custom components every time `mjml_to_html` is called. For my purposes, that's not a big deal, since the call to `mjml_to_html` would be wrapped in another function anyway.